sdk/spring/spring-cloud-azure-feature-management/src/test/java/com/azure/spring/cloud/feature/management/ValidationsTest.java

```diff
@@ -0,0 +1,152 @@
+    }
+
+    @AfterEach
+    public void cleanup() throws Exception {
+        MockitoAnnotations.openMocks(this).close();
+    }
+
+    private boolean hasException(ValidationTestCase testCase) {
+        final String exceptionStr = testCase.getIsEnabled().getException();
+        return exceptionStr != null && !exceptionStr.isEmpty();
+    }
+
+    private boolean hasInput(ValidationTestCase testCase) {
+        final LinkedHashMap<String, Object> inputsMap = testCase.getInputs();
+        return inputsMap != null && !inputsMap.isEmpty();
+    }
+
+    private File[] getFileList(String fileNameFilter) {
+        final URL folderUrl = Thread.currentThread().getContextClassLoader().getResource(testCaseFolderPath);
+        assert folderUrl != null;
+
+        final File folderFile = new File(folderUrl.getFile());
+        final File[] filteredFiles = folderFile.listFiles(pathname -> pathname.getName().toLowerCase().contains(fileNameFilter));
+        assert filteredFiles != null;
+
+        Arrays.sort(filteredFiles, Comparator.comparing(File::getName));
+        return filteredFiles;
+    }
+
+    private List<ValidationTestCase> readTestcasesFromFile(File testFile) throws IOException {
+        final String jsonString = Files.readString(testFile.toPath());
+        final CollectionType typeReference =
+            TypeFactory.defaultInstance().constructCollectionType(List.class, ValidationTestCase.class);
+        return objectMapper.readValue(jsonString, typeReference);
+    }
+
+    @SuppressWarnings("unchecked")
+    private LinkedHashMap<String, Object> readConfigurationFromFile(File sampleFile) throws IOException {
+        final String jsonString = Files.readString(sampleFile.toPath());
+        final LinkedHashMap<String, Object> configurations = objectMapper.readValue(jsonString, new TypeReference<>() {
+        });
+        final Object featureManagementSection = configurations.get("feature_management");
+        if (featureManagementSection.getClass().isAssignableFrom(LinkedHashMap.class)) {
+            return (LinkedHashMap<String, Object>) featureManagementSection;
+        }
+        return new LinkedHashMap<>();
+    }
+
+    @SuppressWarnings("unchecked")
+    private void runTestcases(File sampleFile, File testsFile) throws IOException {
+        // initialize feature manager
+        final FeatureManagementProperties managementProperties = new FeatureManagementProperties();
+        managementProperties.putAll(readConfigurationFromFile(sampleFile));
+        final FeatureManager featureManager = new FeatureManager(context, managementProperties, configProperties);
+
+        final List<ValidationTestCase> testCases = readTestcasesFromFile(testsFile);
+        System.out.println("Running test case from file: " + testsFile.getName());
+
+        for (int i = 0; i < testCases.size(); i++) {
+            System.out.println("Test case " + i + " : " + testCases.get(i).getDescription());
```

Comment:
I don't know if we should bother with the test case number as they are sorted and shouldn't match the files.